### PR TITLE
fix(alert): various fixes

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. fix various issues with rh-alert (see changeset)
2. cherry-picked those fixes from the #1350 for ease of review

### CODE REVIEW

File: elements/rh-alert/rh-alert.css [hunk 1]
```diff
@@ -150,20 +126,46 @@ footer ::slotted([slot="actions"]:hover) {
   color: var(--rh-color-interactive-blue-darkest, #003366) !important;
 }
 
-footer ::slotted(rh-button[variant="link"]) {
+footer ::slotted(rh-button[variant="link" i]) {
```
Comment (zeroedin) [inline elements/rh-alert/rh-alert.css:129]: typo

Comment (bennypowers) [reply elements/rh-alert/rh-alert.css:129]: the `i`? https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors#attr_operator_value_i

File: elements/rh-alert/rh-alert.css [hunk 2]
```diff
@@ -73,19 +43,25 @@ header {
...
 
 header ::slotted(*) {
   font-family: var(--_font-family) !important;
   font-size: var(--rh-font-size-body-text-sm, 0.875rem) !important;
   font-weight: var(--rh-font-weight-heading-bold, 700) !important;
   margin: 0 !important;
+}
+
+header ::slotted(:is(h1,h2,h3,h4,h5,h6)) {
   padding-block: 2px 4px !important;
```
Comment (zeroedin) [inline elements/rh-alert/rh-alert.css:60]: Should this be marked as non standard?

```Suggestion
  padding-block: 2px var(--rh-space-xs, 4px); !important;  /* 2px is a non standard value */
```

File: elements/rh-alert/rh-alert.css (hunk 1 above)
Comment (zeroedin) [reply elements/rh-alert/rh-alert.css:129]: TIL

File: elements/rh-alert/rh-alert.css (hunk 2 above)
Comment (zeroedin) [reply elements/rh-alert/rh-alert.css:60]: This is non blocking

**Review by zeroedin — APPROVED** [review]
Let's Give Timely Messages

File: elements/rh-alert/rh-alert.css [hunk 3]
```diff
@@ -73,19 +43,25 @@ header {
...
 header ::slotted(*) {
   font-family: var(--_font-family) !important;
   font-size: var(--rh-font-size-body-text-sm, 0.875rem) !important;
   font-weight: var(--rh-font-weight-heading-bold, 700) !important;
   margin: 0 !important;
-  padding-block: 2px 4px !important;
+}
+
+header ::slotted(:is(h1,h2,h3,h4,h5,h6)) {
+  padding-block: 2px var(--rh-space-xs, 4px); !important;  /* 2px is a non standard value */
```
Comment (bennypowers) [inline elements/rh-alert/rh-alert.css:60]:
```suggestion
  padding-block: 2px var(--rh-space-xs, 4px) !important;  /* 2px is a non standard value */
```

**Pull request merged**